source/elements/oneDAL/source/appendix/kd_tree.rst

```diff
@@ -0,0 +1,29 @@
+.. _kd_tree:
+
+==================
+Appendix: k-d Tree
```

Comment:
> This is an interesting question. I assume that kd tree topic is in the 'appendix' as it potentially can be referred from multiple sections of the specs.

Is not the only reason. Usually, appendix is "a section at the end of a paper that includes information that is too detailed for the text of the paper itself and would burden the reader". From my point of view, k-d tree is one of those:
- It would be too detailed description in terminology section;
- It's not related to kNN directly;
- There is no user-level k-d tree functionality/API in spec so far to add this to algorithms section (once we have it, it's worth to move from appendix);
- There is no related mathematical notations/formulas for math notation section, even though it had, it would be too detailed.

The plan was the following: 
- Describe it in appendix, add reference to the appendix every time the text mentions k-d tree.
- Add the k-d tree term to the terminology section with the very brief description referring to the appendix.

Reply:
@outoftardis, agree, I'l remove `appendix` from title.

Reply:
OK. @andreynv, what is your recommendation for k-d tree? Where to put it?

Reply:
Looking into overall structure of the specs and assuming that this 'appendix' is a place for description of the common defs/math for the algorithms, I'd put it on the page 'Algorithms' under the first bullet 'xxx', prior 'Clustering' bullet. We also need perspective of @outoftardis to close it

Reply:
I mostly agree with @rlnx here. The current location of appendix is quite logical. I wouldn't put it in the beginning of "Algorithms" section. 
The only time readers would want detailed info about k-d tree is when they encounter it in a description of an algorithm, and as Ruslan says, the plan is to reference this topic whenever k-d is mentioned. What goes into terminology is essential information that users would need to understand the specs. I wouldn't mix these two types together. Which means that this

> Add the k-d tree term to the terminology section with the very brief description referring to the appendix.

might not be a good idea.

I'd propose this:
- add to Terminology topic (1) a paragraph explaining what kind of terms we are including there and (2) a link to the Appendix (the whole chapter) with a note saying what type of info the reader can find there
- add to Appendix topic the same info: what kind of info Appendix contains and that if you are looking for definitions of terms the spec is using, you'd want to check out Terminology topic.

Reply:
Okay, let's have the 'appendix' in the same location
Re the content topic: the desire is to avoid info duplication when possible